File: web/src/types/rebalance.ts
/** 调仓行动项 */
export interface ActionItem {
  fund_code: string
  position_type: string
  action: string
  reason: string
}

/** 调仓结果 */
export interface RebalanceResult {
  as_of: string
  actions: ActionItem[]
  core_picks: [string, number][]
  satellite_picks: [string, number][]
}

File: src/fund_analyzer/api/schemas/rebalance.py
"""调仓建议相关 API 数据模型。"""
from __future__ import annotations

from pydantic import BaseModel


class ActionItem(BaseModel):
    """调仓行动项。"""

    fund_code: str
    position_type: str
    action: str
    reason: str


class RebalanceResult(BaseModel):
    """调仓建议结果。"""

    as_of: str
    actions: list[ActionItem]
    core_picks: list[list]
    satellite_picks: list[list]
